docs/nsm_api.md

```diff
@@ -115,27 +115,48 @@ This means that in the abstract Network Service Mesh has:
 
 The Network Service Manager API is defined in protobuf for communication over grpc.
 
-In this message Client&#39;s local NSM proxying Network Service request to NSE&#39;s local NSM.  2 NSMs need to agree upon a tunneling technology supported by both, hence a list of tunnel types is a part of the request message.
+In this message Client&#39;s local NSM (hereafter called NSM1) proxying Network Service request to NSE&#39;s local NSM (hereafter called NSM2). NSM1 and NSM2 have to negotiate:
+
+<dl>
+    <dt>RemoteConnectionMechanism</dt>
+    <dd>The 'mechanism' (usually, but not necessarily a tunnel type and tunnel parameters) for the L2/L3 connection between the NSMs local dataplanes.</dd>
+    <dt>ConnectionContext<dt>
+    <dd>The L2/L3 Connection has 'context'.  Context includes addressing and routing information **for the L2/L3 connection**.</dd>
+</dl>
+
+The common pattern for negotiation of all of these matters between a NSM1 and NSM2 is:
+1. NSM1 communicates preferences and constraints
+2. NSM2 makes selections which meet those selections and constraints, or rejects the connection.
 
 ```proto
+
+service NSM2NSM {
+    rpc RequestConnection(RemoteConnectionRequest) returns RemoteConnectionResponse;
```

Comment:
`RequestConnection` I think is already used by client/nsm communication. I think we should use a different name.